vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,82 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(string_utils::utf_to_wide(base_path));
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home_dir = getenv("HOME");
+    auto env_data = getenv("XDG_DATA_HOME");
+    if (env_data != NULL)
+        root_paths.set_pref_path(string_utils::utf_to_wide(env_data) / fs::path(app_name) / fs::path(app_name));
+
+    root_paths.set_log_path(root_paths.get_pref_path());
+
+    env_data = getenv("XDG_CONFIG_HOME");
+    if (env_data != NULL)
+        root_paths.set_config_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_config_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".config" / fs::path(app_name));
+
+    env_data = getenv("XDG_CACHE_HOME");
+    if (env_data != NULL)
+        root_paths.set_cache_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".cache" / fs::path(app_name));
+
+    if (env_home_dir != NULL) {
+        // Don't assume that base_path is portable.
+        root_paths.set_shared_path(fs::path(env_home_dir) / ".local/share" / fs::path(app_name));
+    }
+    env_data = getenv("XDG_DATA_DIRS");
```

Comment:
XDG_DATA_DIRS is not always set, if its not set it should use `XDG_DATA_HOME/Vita3K` for the shared path
and if XDG_DATA_HOME is not set defaulting to `~/.local/share`

Reply:
Fixed.